### ISSUE TITLE
Add Windows support and CI

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelicaLibraryTesting#57]
To support library testing on Windows we need to ensure these scripts are running fine on Windows.

**Comment by casella** [comment]
@AndreaBartolini you have some experience with it, can you comment on that?

**Comment by bilderbuchi** [comment]
Windows CI runners are available from Github (e.g. as `windows-latest`): https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Comment by AndreaBartolini** [comment]
I made my modifications starting from the commit 65b4f9db (2023.09.30) and I saw that a lot of commits have been made after that one, so in order to avoid confusion I made a new branch WindowSupport and put my modification on it
After your check and if you agree I can merge my modifications on the master branch and create a pull request.

The only problem is that I have no rights to push on the repo, here the message I got when tryed to push the new branch:

```
$ git push origin WindowsSupport
Enter passphrase for key '/c/Users/andre/.ssh/id_rsa':
ERROR: Permission to OpenModelica/OpenModelicaLibraryTesting.git denied to AndreaBartolini.
fatal: Could not read from remote repository.
```

Can you please give me the rights to push in the repo?

**Comment by AndreaBartolini** [comment]
BTW: on my PC if I run the CI under windows on multiple processes sometimes the last log line is not recorded in the log files. I suppose this is beacuse Python multiprocess is not thread-safe under windows.
I remember that in the past I got a similar issue in Visual C++ under windows.
Running the CI under windows on a single process the problem disappears.

**Comment by bilderbuchi** [comment]
>  suppose this is beacuse Python multiprocess is not thread-safe under windows.

That makes me curious, and would be good to be aware of, do you have further information on this supposition?

**Comment by AndreaBartolini** [comment]
I remember reading something about this on python documentation online when I investigated the problem of the missing line in the log file, unfortunately I've not saved the  page address...

What I've understood is that this problem arises when omc fails due to the long path of the model name, rising the error:

`[C:/dev/OM64bit/OMCompiler/Compiler/SimCode/SimCodeMain.mo:532:7-532:78:writable] Error: Internal error C:\Program Files\OpenModelica1.23.0-dev-64bit\bin\omc.exe: [unknown function name] failed`

This error line is missing in the log file if I run more than one process in parallel (--jobs = zero or higher than 1), so I suppose that the problem is the thread termination before the file writing is completed when more than one thread are running in parallel.

I've not found a way to better investigate the problem, but is seems to be structural because it happens anytime omc raises said error. Maybe it depends on the way the omc exits when this error arises, but I don't know omc so deeply to verify this.

The problem is windows related becase under linux there is no path limitation.

**Comment by AnHeuermann** [comment]
Sure.

First create a fork of this repository:

![grafik](https://github.com/OpenModelica/OpenModelicaLibraryTesting/assets/38031952/adc528d0-f82a-42c4-b75d-e9adbc0cb61e)

https://github.com/OpenModelica/OpenModelicaLibraryTesting/fork

That is your personal copy AndreaBartolini/OpenModelicaLibraryTesting where you can push your new branch `WindowsSupport` with the changes you want to contribute to OpenModelica/OpenModelicaLibraryTesting.

Then create a new pull request:

![grafik](https://github.com/OpenModelica/OpenModelicaLibraryTesting/assets/38031952/7809c0c7-c925-46ea-9927-ee423265db89)

https://github.com/OpenModelica/OpenModelicaLibraryTesting/compare

**Comment by AndreaBartolini** [comment]
the PR #58 partially implements the windows support
